app/helpers/admin/users_helper.rb

```diff
@@ -89,5 +89,25 @@ def organization_tooltip(organization_names, count, imposed_limit: 0)
         str + " & #{overflow_count(count, imposed_limit: imposed_limit)} others"
       end
     end
+
+    # Returns a boolean that indicates whether a user can be banished
+    # A user can be banished when:
+    # - the users account was created in the period before the last 100 days ago
+    # and there are no comments from the user in the period before the last 100 days.
+    # - if the logged in user is a super admin or support admin
+
+    # Some practical scenarios:
+    # Today is the 06 July:
+    # - if a user was created before the 27 March 2022 (101 days ago)
+    # and they had no comments before the 27 March 2022 then they can be banished.
+    # - if they have any comments before the 27 March 2022 then they cannot be banished
+    # - if they were created before the 27 March 2022 (101 days ago) and they have comments
+    # before the 27 March 2022 then they cannot be banished.
+    #  @param {Array} [array] the user
+    # @return [Boolean]
+    def banishable_user?(user)
+      (user.comments.where("created_at < ?", 100.days.ago).empty? && user.created_at < 100.days.ago) ||
```

Comment:
As written this method queries the database using 4 separate trips.  It may be adequate for the show page, but will potentially be an issue for the index page.

Consider favoring `user.last_commented_at` instead of querying the user's comments.

(See https://github.com/forem/forem/blob/64dd92d60d93a1543253828fd9a54f7b39f99c2e/db/schema.rb#L1234)

Reply:
Definitely see a performance concern, however it feels a bit weird that our default `last_commented_at` is set to `"2017-01-01 05:00:00"` even if the comments array is []. Hence, my check would need to revolve around this last_commented_at hardcoded date. 

Thoughts?